blockade/api/manager.py

```diff
@@ -14,59 +14,95 @@
 # limitations under the License.
 #
 
+from blockade.config import BlockadeConfig
 from blockade.core import Blockade
+from blockade.errors import BlockadeNotFound
 from blockade.errors import InvalidBlockadeName
 from blockade.net import BlockadeNetwork
 from blockade.state import BlockadeState
 
-# TODO(pdmars): breaks if server restarts, refactor to be part of BlockadeState
-BLOCKADE_CONFIGS = {}
+import os
+import yaml
+
+
 DATA_DIR = "/tmp"
+BASE_BLOCKADE_DIR = os.path.join(DATA_DIR, ".blockade")
+REST_STATE_FILE = os.path.join(BASE_BLOCKADE_DIR, "rest_state.yaml")
 
 
 class BlockadeManager:
-    """Simple helper for what should eventually be persisted via BlockadeState
+    """Simple helper to persist Blockade configurations managed via REST API
     """
 
     @staticmethod
     def set_data_dir(data_dir):
         global DATA_DIR
         DATA_DIR = data_dir
+        BASE_BLOCKADE_DIR = os.path.join(DATA_DIR, ".blockade")
+        REST_STATE_FILE = os.path.join(BASE_BLOCKADE_DIR, "rest_state.yaml")
+
+
+    @staticmethod
+    def init_base_blockade_dir():
+        if not os.path.isdir(BASE_BLOCKADE_DIR):
+            os.makedirs(BASE_BLOCKADE_DIR)
+
+    @staticmethod
+    def read_rest_state():
+        BlockadeManager.init_base_blockade_dir()
+        rest_state = {}
+        if os.path.exists(REST_STATE_FILE):
+            with open(REST_STATE_FILE, "r") as f:
+                rest_state = yaml.safe_load(f) or {}
+        return rest_state
+
+    @staticmethod
+    def write_rest_state(rest_state):
+        BlockadeManager.init_base_blockade_dir()
+        with open(REST_STATE_FILE, "w") as f:
+            yaml.safe_dump(rest_state, f)
 
     @staticmethod
     def blockade_exists(name):
-        global BLOCKADE_CONFIGS
-        return name in BLOCKADE_CONFIGS
+        rest_state = BlockadeManager.read_rest_state()
+        return name in rest_state
 
     @staticmethod
     def store_config(name, config):
-        global BLOCKADE_CONFIGS
-        BLOCKADE_CONFIGS[name] = config
+        rest_state = BlockadeManager.read_rest_state()
+        rest_state[name] = config
+        BlockadeManager.write_rest_state(rest_state)
```

Comment:
I'm not too familiar with Flask - are there multiple request handler threads? If so this behavior seems racy. Concurrent requests could step on each other. `delete_config` too.